demos/link-prediction/knowledge-graphs/complex.ipynb

```diff
@@ -0,0 +1,708 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "wn18_raw_ranks = ComplEx.rank_edges_against_all_nodes(\n",
+    "    wn18_model, wn18_gen.flow(wn18_test), wn18_g\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# helper function to compute metrics from a dictionary of name -> array of ranks\n",
+    "def summarise(name_to_ranks):\n",
+    "    return pd.DataFrame(\n",
+    "        [\n",
+    "            {\n",
+    "                \"mrr\": np.mean(1 / ranks),\n",
+    "                \"hits at 1\": np.mean(ranks <= 1),\n",
+    "                \"hits at 3\": np.mean(ranks <= 3),\n",
+    "                \"hits at 10\": np.mean(ranks <= 10),\n",
+    "            }\n",
+    "            for ranks in name_to_ranks.values()\n",
+    "        ],\n",
+    "        index=name_to_ranks.keys(),\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "<div>\n",
+       "<style scoped>\n",
+       "    .dataframe tbody tr th:only-of-type {\n",
+       "        vertical-align: middle;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe tbody tr th {\n",
+       "        vertical-align: top;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe thead th {\n",
+       "        text-align: right;\n",
+       "    }\n",
+       "</style>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>mrr</th>\n",
+       "      <th>hits at 1</th>\n",
+       "      <th>hits at 3</th>\n",
+       "      <th>hits at 10</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>raw</th>\n",
+       "      <td>0.571233</td>\n",
+       "      <td>0.4501</td>\n",
+       "      <td>0.6468</td>\n",
+       "      <td>0.8026</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "          mrr  hits at 1  hits at 3  hits at 10\n",
+       "raw  0.571233     0.4501     0.6468      0.8026"
+      ]
+     },
+     "execution_count": 11,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "summarise({\"raw\": wn18_raw_ranks})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For comparison, Table 2 in the paper gives the following results for WN18:\n",
+    "\n",
+    "| raw MRR | filtered MRR | hits at 1 | hits at 3 | hits at 10 |\n",
```

Comment:
I think we should mention that these are filtered hits

```suggestion
    "| raw MRR | filtered MRR | filtered hits at 1 | filtered hits at 3 | filtered hits at 10 |\n",
```

Reply:
I've combined this with the suggestion of including these results as a dataframe, to have it have the raw and filtered numbers on separate rows.